pkg/cmd/server/start/start_node.go

```diff
@@ -255,22 +253,25 @@ func RunSDNController(config *kubernetes.NodeConfig, nodeConfig configapi.NodeCo
 	if err != nil {
 		glog.Fatal("Failed to get kube client for SDN")
 	}
-	registry := osdn.NewOsdnRegistryInterface(oclient, config.Client)
-
-	switch nodeConfig.NetworkConfig.NetworkPluginName {
-	case flatsdn.NetworkPluginName():
-		ch := make(chan struct{})
-		config.KubeletConfig.StartUpdates = ch
-		go flatsdn.Node(registry, nodeConfig.NodeName, nodeConfig.NodeIP, ch, nodeConfig.NetworkConfig.MTU)
-	case multitenant.NetworkPluginName():
-		ch := make(chan struct{})
-		config.KubeletConfig.StartUpdates = ch
-		plugin := multitenant.GetKubeNetworkPlugin()
-		config.KubeletConfig.NetworkPlugins = append(config.KubeletConfig.NetworkPlugins, plugin)
-		go multitenant.Node(registry, nodeConfig.NodeName, nodeConfig.NodeIP, ch, plugin, nodeConfig.NetworkConfig.MTU)
-		return registry
+
+	ch := make(chan struct{})
+	config.KubeletConfig.StartUpdates = ch
+
+	controller, endpointFilter, err := factory.NewPlugin(nodeConfig.NetworkConfig.NetworkPluginName, oclient, config.Client, nodeConfig.NodeName, nodeConfig.NodeIP, ch)
```

Comment:
Previously, we only called our constructor for specific recognized network plugin names. Now we assume all network plugin names are meant for our constructor. Is that the case, or are there upstream network plugins someone could invoke that we should no-op for?